src/components/Table/TableHead/TableHead.jsx

```diff
@@ -149,6 +149,33 @@ const TableHead = ({
   i18n,
 }) => {
   const filterBarActive = activeBar === 'filter';
+  let isResizable = false;
+  let startWidth = 0;
+  let startX = 0;
+  let currentElement = '<>';
+  const onMouseMovecallback = e => {
+    if (isResizable) {
+      currentElement.style.width = `${startWidth + (e.pageX - startX)}px`;
```

Comment:
The width shouldn't be set imperatively like this. We need to store the width of each `TableHeader` in state via an object keyed with the same key placed on `TableHeader`, `column-${matchingColumnMeta.id}`. This would be done in a useState/useEffect hook. Then we would update the state when the handler/resize slider moves. 
Each `TableHeader` would have 
```js
style={{width: this.state.tableHeaders[{`column-${matchingColumnMeta.id}`}]}}`
```

Unfortunately to do this, we need to access the width of the `<th>` in TableHeader via a `ref` in the `useEffect` hander for each `TableHeader`. This is currently not possible. Unfortunately the `TableHeader` component is provided by Carbon, we don't control the source, and it is a functional component. Function components do not accept the `ref` attribute. Carbon can forward the `ref`, which would enable us to then use the refs to do this appropriately.

[I've put a PR in on Carbon to update TableHeader to forward it's ref.](https://github.com/carbon-design-system/carbon/pull/4877) 

Until that's been merged, published, and we update to that version we seem to be blocked. In the meantime we can temporarily completely reimplement `TableHeader` to forward the ref while we wait on the contribution to be accepted.

Reply:
@tay1orjones Initially we tried with react hooks (useEffect, useState)  but we were not able to get the reference of specific column, and while setting/resizing the width, width was setting for all table columns. 

So what we understand that you have added the forward ref changes in carbon and we are waiting for PR acceptance (which you raised) and Carbon-IOT version upgrade (with latest Carbon) before adding our table header resize changes.

Once version upgrade is done in carbon-iot then we will implement your suggestion of getting individual <th> ref width using useEffect and storing the width of each TableHeader using useState and applying resized width (style={{width: this.state.tableHeaders[{`column-${matchingColumnMeta.id}`}]}}`) .  

Please correct me if our understanding is not in sync.

Reply:
Yep, that's correct. In the meantime, you can reimplement/copy `TableHeader` from my carbon PR and use that for the time being. This way you're not blocked and can work through the rest of the implementation.

Depending on the timeline and our ability to update to the new version that carbon will release with my change, we may wait to merge your changes until we've updated. Another option would be to merge your changes with the reimplemented/copied `TableHeader` with a new issue created to refactor once we've updated the Carbon version.

Reply:
@davidicus PR #788 is updated for ref created in Carbon, here we are using ref for <th> reference using react hooks as suggested by @tay1orjones . 
In carbon a ref={ref}) is created/forward to use in Carbon-IOT, but we also observed that same is also missing if sorting is false on a column, which we have added in TableHeader.js, this also need to be added in Carbon.

 Please review the code.